playground/nextjs-app-router/lib/useReservoirMintData.ts

```diff
@@ -0,0 +1,147 @@
+import type { ContractType } from '@/onchainkit/esm/nft/types';
+import type { definitions } from '@reservoir0x/reservoir-sdk';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { ENVIRONMENT_VARIABLES } from './constants';
+
+type NonNullable<T> = T & {};
+type Collection = NonNullable<
+  definitions['getCollectionsV7Response']['collections']
+>[0];
+type Token = NonNullable<
+  NonNullable<definitions['getTokensV7Response']['tokens']>[0]
+>['token'];
+
+function useToken(contractAddress: string, tokenId?: string) {
+  return useQuery({
+    queryKey: ['token', contractAddress, tokenId],
+    queryFn: async () => {
+      const qs = tokenId
+        ? `tokens=${contractAddress}:${tokenId}`
+        : `collection=${contractAddress}`;
+      const response = await fetch(
+        `https://api-base.reservoir.tools/tokens/v7?${qs}&includeLastSale=true`,
+        {
+          method: 'GET',
+          headers: {
+            accept: '*/*',
+            'x-api-key': ENVIRONMENT_VARIABLES.RESERVOIR_API_KEY ?? '',
+          },
+        },
+      );
+      const data = await response.json();
+      // if no tokenId, get the collection and default to the first token
+      return data.tokens[0].token as Token;
+    },
```

Comment:
little confused by this comment. should we first try to grab by tokenId before grabbing the first token?

Reply:
this endpoint is set up to accept tokens=ContractAddress:TokenId or collection=ContractAddress.  If no tokenId is passed in, I call with collection which returns all the tokens in the collection, and I just pick the first one.  It'd most likely be the same as calling with tokenId = 0, but also would handle weird contracts that started at 1